.github/workflows/ci.yml

```diff
@@ -1,4 +1,15 @@
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - main
+
+  pull_request:
+    branches:
+      - '**'
+
+permissions:
+  contents: read
+  pages: write
```

Comment:
Maybe it's a nit but can you scope the permission only where required?

`zizmor` reports the same:

```
error[excessive-permissions]: overly broad permissions
  --> ./.github/workflows/ci.yml:12:3
   |
12 |   pages: write
   |   ^^^^^^^^^^^^ pages: write is overly broad at the workflow level
   |
   = note: audit confidence → High
```

Reply:
Done.  I also fixed the other `zizimor` complaint, about persisting credentials on checkout